source/MaterialXRuntime/RtGeneric.cpp

```diff
@@ -37,12 +38,22 @@ RtPrim RtGeneric::createPrim(const RtToken& typeName, const RtToken& name, RtPri
 const RtToken& RtGeneric::getKind() const
 {
     RtTypedValue* v = prim()->getMetadata(KIND);
-    return v->getValue().asToken();
+    return v && v->getType() == RtType::TOKEN ? v->getValue().asToken() : UNKNOWN;
```

Comment:
If the prim was created by another schema the "kind" metadata might not exists, so need to handled that here.